### ISSUE TITLE
Weird behavior (model trained on a custom dataset)

### ISSUE THREAD

**Issue by zyang37** [yassouali/CCT#13]
Hello, Thank you so much for the great work.

I am trying to train it on a custom dataset which contains 7 classes, we assigned pixel value 1-7 to each classes and 0 as background. We generated some labels (shown below), it looks like a black image but the pixel value for the object should not be 0. And I also built a custom dataloader for it. 
![b4900225 17](https://user-images.githubusercontent.com/47701932/94203753-f99ed000-fe8d-11ea-9796-5cb2bd7ed4fe.png)

After training, here is a sample prediction and I am not sure what causes this. The black parts in the middle is where the object was located, the model is literally labeling the background for some reason. 
![2b701108](https://user-images.githubusercontent.com/47701932/94203154-e4757180-fe8c-11ea-848e-9fabe2b448d7.png)

Is the labeling image correct?

**Comment by yassouali** [comment]
Hi @zyang37 
Thank you for your interest; 

So if I understood correctly, the problem is the model's prediction or the input data, or both?

For your dataset, with 8 classes in total, 0-7 with the background included, so during training, the model is in fact trained on the background classes and is capable of predicting it. If you want to only train on the foreground classes, you can set `ignore_label` in the cross-entropy loss. But before training, make sure the labels are loaded correctly.

**Comment by akmaralAW** [comment]
Hello @zyang37,

You mentioned "we assigned pixel value 1-7 to each classes and 0 as background." Where these values should be adjusted ?

**Comment by zyang37** [comment]
Hi @akmaralAW, 
I am also still working on it. But as far as i know 0-7 are actual pixel values. These values corresponds to classes. The region where object (class) 1 stands, the pixel values at that region should all be 1, same for other classes.

**Comment by akmaralAW** [comment]
Thanks for your response, @zyang37 

Is it in annotation part ? Can you give me some image and corresponding pixel value please.

**Comment by zyang37** [comment]
Hello, 
You can find them in VOC2012.
